src/riak_test_escript.erl

```diff
@@ -236,8 +236,54 @@ parse_command_line_tests(ParsedArgs) ->
         end, [], lists:usort(DirTests ++ SpecificTests)).
 
 extract_test_names(Test, {CodePaths, TestNames}) ->
-    {[filename:dirname(Test) | CodePaths],
-     [filename:rootname(filename:basename(Test)) | TestNames]}.
+
+    RTTestPaths = get_test_paths(),
+    CommaSepTests = string:tokens(Test, [$,]),
+
+    WCTestsNoPrefix = [get_test_with_valid_prefix(TestA, RTTestPaths) || TestA <- CommaSepTests, get_test_type(TestA) =:= yes_wc_no_prefix],
+    WCTestsPrefix = [TestA || TestA <- CommaSepTests, get_test_type(TestA) =:= yes_wc_yes_prefix],
+    NotWCTestsNoPrefix = [get_test_with_valid_prefix(TestA, RTTestPaths) || TestA <- CommaSepTests, get_test_type(TestA) =:= no_wc_no_prefix],
+    NotWCTestsPrefix = [TestA || TestA <- CommaSepTests, get_test_type(TestA) =:= no_wc_yes_prefix],
+
+    AllWCTests = lists:append([filelib:wildcard(TestC) || TestC <- WCTestsNoPrefix++WCTestsPrefix]),
+    AllNotWCTests = NotWCTestsNoPrefix++NotWCTestsPrefix,
+    AllTests = AllWCTests++AllNotWCTests,
+
+    CodePathList = [filename:dirname(TestName) || TestName <- AllTests],
+    TestNameList = [filename:rootname(filename:basename(TestName)) || TestName <- AllTests],
+    {CodePathList++CodePaths, TestNameList++TestNames}.
+
+get_test_type(Test) ->
+    case string:str(Test, "*") of
+        0 ->
+          case string:str(Test, "/") of
+              0 -> no_wc_no_prefix;
+              _ ->  no_wc_yes_prefix
+          end;
+        _ ->
+          case string:str(Test, "/") of
+              0 -> yes_wc_no_prefix;
+              _ -> yes_wc_yes_prefix
+          end
+    end.
+
+get_test_paths() ->
+    RTTestPaths = rt_config:get(test_paths, ["tests"]),
```

Comment:
How about we simply have a default value of `[]` and return the value of `rt_config:get/2` concatenated with `tests`?

`[rt_config:get(test_paths, [])|["tests"]]`